Django_web/myfirst/apps/articles/views.py

```diff
@@ -0,0 +1,36 @@
+from django.http import Http404, HttpResponseRedirect, HttpResponse
+from django.shortcuts import render
+from django.urls import reverse
+
+from .models import Article
+
+
+def home(request):
+    return render(request, 'articles/home_list.html')
+
+
+def index(request):
+    latest_articles_list = Article.objects.order_by('-pub_date')[:5]  # вернем последние 5 статей
+    return render(request, 'articles/list.html', {'latest_articles_list': latest_articles_list})
+
+
+def detail(request, article_id):
+    try:
+        a = Article.objects.get(id=article_id)
+    except:
+        raise Http404("Статья не найдена :(")
+
+    latest_comments_list = a.comment_set.order_by('-id')[:10]
+
+    return render(request, 'articles/detail.html', {'article': a, 'latest_comments_list': latest_comments_list})
+
+
+def leave_comment(request, article_id):
+    try:
+        a = Article.objects.get(id=article_id)
+    except:

```

Comment:
Too broad exception, probably need to use Article.[DoesNotExist ](https://docs.djangoproject.com/en/3.2/ref/models/class/#django.db.models.Model.DoesNotExist)

Reply:
Yes, changed